packages/shared/src/UI/components/AddCollectibles/index.tsx

```diff
@@ -0,0 +1,300 @@
+    const { pluginID, chainId, account: defaultAccount, onClose, onConfirmLoading } = props
+    const t = useSharedI18N()
+    const walletAccount = useAccount()
+    const account = defaultAccount || walletAccount
+    const { classes } = useStyles(undefined, { props })
+    const Others = useWeb3Others(pluginID)
+
+    const {
+        control,
+        watch,
+        handleSubmit,
+        resetField,
+        formState: { errors, isValidating },
+    } = useForm({
+        mode: 'all',
+        defaultValues: { address: '', tokenIds: '' },
+    })
+    const watchedTokenIds = watch('tokenIds')
+    const tokenIds = useMemo(() => uniq(compact(watchedTokenIds.split(',').map((id) => id.trim()))), [watchedTokenIds])
+    const address = watch('address')
+    const hub = useWeb3Hub(pluginID, { chainId })
+    const connection = useWeb3Connection(pluginID, { chainId })
+
+    const validationMsgForAddress = useMemo(() => {
+        if (address && !Others.isValidAddress?.(address)) return t.collectible_contract_invalid()
+        return ''
+    }, [address])
+
+    const {
+        data: contract,
+        isLoading: isLoadingContract,
+        isError,
+        refetch,
+    } = useQuery({
+        queryKey: ['nft-contract', pluginID, chainId, address],
+        queryFn: () => connection.getNonFungibleTokenContract(address, undefined, { chainId }),
+    })
+
+    const isValid = useMemo(() => {
+        return Boolean(
+            contract?.address && Others.isValidAddress?.(contract?.address) && isValidTokenIds(watchedTokenIds),
+        )
+    }, [contract?.address, watchedTokenIds])
+
+    const assetsQueries = useQueries({
+        queries: tokenIds.map((tokenId) => ({
+            enabled: isValid,
+            queryKey: ['nft-asset', pluginID, chainId, address, tokenId],
+            queryFn: () => hub.getNonFungibleAsset(address, tokenId, { chainId }),
+        })),
+    })
+    const noResults = assetsQueries.every((x) => !x.isLoading && !x.data)
+    const someNotMine = assetsQueries.some((x) => (x.data ? !isSameAddress(x.data.owner?.address, account) : false))
+
+    const handleFormSubmit = useCallback(
+        (event: FormEvent<HTMLFormElement>) => {
+            handleSubmit(() => {})(event)
+        },
+        [handleSubmit],
+    )
+    const [selectedTokenIdsMap, setSelectedTokenIdsMap] = useState<Record<string, string[]>>({})
+    const selectedTokenIds = selectedTokenIdsMap[address] || EMPTY_LIST
+
+    const toggleSelect = useCallback(
+        (asset: Web3Helper.NonFungibleAssetAll) => {
+            setSelectedTokenIdsMap((idsMap) => {
+                const ids = idsMap[address] ?? []
+                const newIds = ids.includes(asset.tokenId)
+                    ? ids.filter((x) => x !== asset.tokenId)
+                    : [...ids, asset.tokenId]
+                return {
+                    ...idsMap,
+                    [address]: newIds,
```

Comment:
```suggestion
                    [formatAddress(address)]: newIds,
```

Reply:
It will cause inconsistent results when an unformatted address is used as a key.